src/main/java/org/motechproject/mots/mapper/ModuleAssignmentMapper.java

```diff
@@ -41,24 +51,25 @@ default CommunityHealthWorker toChw(String chwId) {
    * @param moduleId id of Module.
    * @return Module with given id
    */
-  default Module toModule(String moduleId) {
+  protected Module toModule(String moduleId) {
     if (moduleId == null || moduleId.isEmpty()) {
       return null;
     }
 
     return new Module(UUID.fromString(moduleId));
   }
 
-  /**
-   * Get module id as String.
-   * @param module module assigned to CHW
-   * @return module id as String
-   */
-  default String fromModule(Module module) {
-    if (module == null || module.getId() == null) {
-      return null;
+  @AfterMapping
+  protected void fillIsStared(AssignedModules assignedModules,
+      @MappingTarget ChwModulesDto chwModulesDto) {
+    UUID chwId = assignedModules.getHealthWorker().getId();
+    for (ModuleSimpleDto module : chwModulesDto.getModules()) {
+      module.setIsStarted(isModuleStarted(chwId, UUID.fromString(module.getId())));
     }
+  }
 
-    return module.getId().toString();
+  private Boolean isModuleStarted(UUID chwId, UUID moduleId) {
+    return moduleProgressService.getModuleProgress(chwId, moduleId).getStatus()
```

Comment:
you should use !ProgressStatus.NOT_STARTED.equals(moduleProgressService.getModuleProgress(chwId, moduleId).getStatus())

Reply:
Maybe it's worth adding isStarted() method to the ProgressStatus enum?

Reply:
yes, that's a good idea

Reply:
Yes, I think it is a good idea. I will change it